conans/client/configure_environment.py

```diff
@@ -53,10 +53,44 @@ def __init__(self, *args):
         except:
             self.libcxx = None
 
+    def _gcc_arch_flags(self):
+        if self.arch == "x86_64":
+            return "-m64"
```

Comment:
At least I'm consistent :)
Joking aside, as for CMake I don't think it can hurt to be explicit about bitness depending on `arch`.
The only regression would be if some existing package was not built on 64 bits with `arch=x86_64`, which I'd prefer to call a bug.